compendium/DeclarativeServices/src/ComponentContextImpl.cpp

```diff
@@ -80,12 +80,13 @@ namespace cppmicroservices
                                       auto bc = GetBundleContext();
                                       auto boundServicesCacheHandle = boundServicesCache.lock();
                                       auto& serviceMap = (*boundServicesCacheHandle)[refName];
+                                      // get us the service instance
                                       cppmicroservices::ServiceObjects<void> sObjs = bc.GetServiceObjects(sRefU);
                                       auto interfaceMap = sObjs.GetService();
-                                      if (interfaceMap)
+                                      if (interfaceMap && interfaceMap->size() != 0)
```

Comment:
Consider "!interfaceMap->empty()" as more descriptive that comparing size to zero.